SU2_CFD/src/interfaces/cfd/CSlidingInterface.cpp

```diff
@@ -47,24 +47,18 @@ void CSlidingInterface::GetDonor_Variable(CSolver *donor_solution, CGeometry *do
                                           const CConfig *donor_config, unsigned long Marker_Donor,
                                           unsigned long Vertex_Donor, unsigned long Point_Donor) {
 
-  unsigned short iVar, nDonorVar;
-  nDonorVar = donor_solution->GetnPrimVar();
-
-  /*---  the number of primitive variables is set to two by default for the turbulent solver ---*/
-  bool turbulent = (nDonorVar == 2) ;
-
+  const unsigned short nDonorVar = donor_solution->GetnPrimVar();
+  const bool turbulent = (nDonorVar==1) || (nDonorVar==2); // SA or SST
```

Comment:
```suggestion
  const auto nDonorVar = donor_solution->GetnPrimVar();
  /// TODO: Replace with approach compatible with any number of variables (e.g. encapsulate in a "solver info" object).
  const bool turbulent = nDonorVar <= 2;
```

Reply:
Yes, `GetnPrimVar()` seems to be non-zero on all flow and turbulence solvers.